mars-form/src/Component/form.js

```diff
@@ -0,0 +1,372 @@
+import React , {Component} from "react";
+import countries from "./countries"
+
+
+class Form extends Component{
+    state =  {
+        firstName: "",
+        lastName: "",
+        dateOfBirth: "",
+        countries: countries,
+        country: "",
+        dietPref: "omnivore",
+        whyMars: "",
+        formCompleted: false,
+
+    }
+    handleFirstName = e =>{
+        this.setState({
+            firstName: e.target.value
+        })
+    }
+    handleLastName = e =>{
+        this.setState({
+            lastName: e.target.value
+        })
+    }
+    handleDateOfBirth = e =>{
+        this.setState({
+            dateOfBirth: e.target.value
+        })
+    }
+    handleSelect = e =>{
+        e.preventDefault()
+        this.setState({
+            country: e.target.value
+        })
+    }
+   
+    populateSelect = () =>{   
+     return this.state.countries.map(country =>{
+         return <option value={country.name} key ="i">{country.name}</option>
+     }) 
+    }
+    handleDiet = e =>{
+        e.preventDefault()
+        this.setState({
+            dietPref: e.target.value
+        })
+    }
+
+    handleSubmit = e => {
+        e.preventDefault();
+        let { firstName, lastName, dietPref, dateOfBirth, country, whyMars} = this.state;
+        console.log({
+            firstName: firstName,
+            lastName: lastName,
+            dateOfBirth: dateOfBirth,
+            option:dietPref,
+            country: country,
+            whyMars: whyMars,
+            onChange:this.handleChange
+        })
+    }
+    handleMars = e => {
+        
+        this.setState({
+             whyMars: e.target.value
+        })
+    }
+    handleChange = e => {
+        //this is the outline to handle the radio buttons
+        this.setState({
+            [e.target.name]:e.target.value
+        })
+    }
+
+    render(){
+        console.log(this.state.whyMars)
+
+        
```

Comment:
All of these inputs should have their value being controlled by state. Go back and revisit the class example. value={this.state.firstName} . We want to have controlled inputs. Same goes for radio buttons and the rest.